website/docs/r/spring_cloud_app_cosmosdb_association.html.markdown

```diff
@@ -0,0 +1,104 @@
+---
+subcategory: "Spring Cloud"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_spring_cloud_app_cosmosdb_association"
+description: |-
+  Associates a [Spring Cloud Application](spring_cloud_app.html) with a [CosmosDB Account](cosmosdb_account.html).
+---
+
+# azurerm_spring_cloud_app_cosmosdb_association
+
+Associates a [Spring Cloud Application](spring_cloud_app.html) with a [CosmosDB Account](cosmosdb_account.html).
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_spring_cloud_service" "example" {
+  name                = "example-springcloud"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+resource "azurerm_spring_cloud_app" "example" {
+  name                = "example-springcloudapp"
+  resource_group_name = azurerm_resource_group.example.name
+  service_name        = azurerm_spring_cloud_service.example.name
+}
+
+resource "azurerm_cosmosdb_account" "example" {
+  name                = "example-cosmosdb-account"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  offer_type          = "Standard"
+  kind                = "GlobalDocumentDB"
+
+  consistency_policy {
+    consistency_level = "Strong"
+  }
+
+  geo_location {
+    location          = azurerm_resource_group.example.location
+    failover_priority = 0
+  }
+}
+
+resource "azurerm_spring_cloud_app_cosmosdb_association" "example" {
+  name                = "example-bind"
+  spring_cloud_app_id = azurerm_spring_cloud_app.example.id
+  cosmosdb_account_id = azurerm_cosmosdb_account.example.id
+  api_type            = "table"
+  cosmosdb_access_key = azurerm_cosmosdb_account.example.primary_key
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Spring Cloud Application Association. Changing this forces a new resource to be created.
+
+* `spring_cloud_app_id` - (Required) Specifies the ID of the Spring Cloud Application where this Association is created. Changing this forces a new resource to be created.
+
+* `cosmosdb_account_id` - (Required) Specifies the ID of the CosmosDB Account. Changing this forces a new resource to be created.
+
+* `api_type` - (Required) Specifies the api type which should be used when connecting to the CosmosDB Account. Possible values are `cassandra`, `gremlin`, `mongo`, `sql` or `table`. Changing this forces a new resource to be created.
```

Comment:
couldn't we determine this from the supplied `cosmosdb_x` property? and then remove this

Reply:
yes, we could do that. I am just afraid that `table` type does not need to supply those optional property. It may be not easy for users to understand the difference

Reply:
does it make sense to even allow specifying type without the cosmosdb property? and the api doesn't error when they are all left empty?

Reply:
for all api type, cosmosDB account is required. For api type `table`, no more property is needed. For the rest api type, Additional property needs to be provided

Reply:
ahh that makes sense - thanks